Needs update for QuickCheck 2.7

Hi,

I’m getting build failures on Debian:
```
Preprocessing test suite 'properties' for ed25519-0.0.2.0...
[1 of 1] Compiling Main             ( tests/properties.hs, dist-ghc/build/properties/properties-tmp/Main.o )

tests/properties.hs:90:9:
    Constructor `Failure' should have 10 arguments, but has been given 8
    In the pattern: Failure n _ _ _ _ _ _ _
    In a case alternative: Failure n _ _ _ _ _ _ _ -> return (False, n)
    In a stmt of a 'do' block:
      case r of {
        Success n _ _ -> return (True, n)
        GaveUp n _ _ -> return (True, n)
        Failure n _ _ _ _ _ _ _ -> return (False, n)
        _ -> return (False, 0) }
make: *** [build-ghc-stamp] Error 1
```
http://aws-logs.debian.net/ftbfs-logs/2014/08/30/haskell-ed25519_0.0.2.0-2_unstable.log

It seems that you need to upgrade to the latest QuickCheck.

--- a/tests/properties.hs
+++ b/tests/properties.hs
@@ -86,7 +86,9 @@ tests ntests =
       case r of
         Success n _ _           -> return (True, n)
         GaveUp  n _ _           -> return (True, n)
-#if MIN_VERSION_QuickCheck(2,6,0)
+#if MIN_VERSION_QuickCheck(2,7,0)
+        Failure n _ _ _ _ _ _ _ _ _ -> return (False, n)
+#elif MIN_VERSION_QuickCheck(2,6,0)
         Failure n _ _ _ _ _ _ _ -> return (False, n)
 #else
         Failure n _ _ _ _ _ _   -> return (False, n)

